### PR TITLE
[Android] gradle copy resources to assets folder before `preDebug|ReleaseBuild`

### PR DESCRIPTION
**Pull request by PatriceJiang**
ref https://github.com/cocos2d/cocos2d-x/pull/19703#issuecomment-493297998

### CODE REVIEW

File: templates/cpp-template-default/proj.android/app/build.gradle [hunk 1]
```diff
@@ -111,19 +111,22 @@ android {
 }
 
 android.applicationVariants.all { variant ->
-    def project_root_folder = "${buildDir}/../../.."
-    def dest_assets_folder = "${buildDir}/../assets"
+    def project_root_folder = "${projectDir}/../.."
+    def dest_assets_folder = "${projectDir}/assets"
```
Comment (minggo) [inline templates/cpp-template-default/proj.android/app/build.gradle:115]: What's the value of `projectDir `?

File: templates/cpp-template-default/proj.android/app/build.gradle [hunk 2]
```diff
@@ -111,19 +111,22 @@ android {
...
+    def copyTaskName = "copy${targetName}ResourcesToAssets"
+    
+    tasks.register(copyTaskName) {
         copy {
             from project_root_folder + "/Resources"
             into dest_assets_folder
             exclude "**/*.gz"
         }
     }
+    tasks.getByName("pre${targetName}Build").dependsOn copyTaskName
```
Comment (minggo) [inline templates/cpp-template-default/proj.android/app/build.gradle:129]: What does it mean?

File: templates/cpp-template-default/proj.android/app/build.gradle (hunk 1 above)
Comment (PatriceJiang) [reply templates/cpp-template-default/proj.android/app/build.gradle:115]: According to https://docs.gradle.org/current/userguide/writing_build_scripts.html


>Name | Type | Default Value
>-- | -- | --
>projectDir | File | The directory containing the build script.

File: templates/cpp-template-default/proj.android/app/build.gradle (hunk 2 above)
Comment (PatriceJiang) [reply templates/cpp-template-default/proj.android/app/build.gradle:129]: Schedule task `copyDebug(Release)ResourcesToAsset` to run before `preDebug(Release)Build`, which will always run ahead of build.

**Pull request merged**